### PR TITLE
fix: keep old behaviour of `json()` by default

### PR DESCRIPTION
**Pull request by PrettyWood**
## Change Summary
fix regression on 1.9 introduced in #2650 (see https://github.com/samuelcolvin/pydantic/discussions/3538#discussioncomment-1843005)

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [ ] ~Documentation reflects the changes where applicable~
* [ ] ~`changes/<pull request or issue id>-<github username>.md` file added describing change~ (I didn't add a change file since it's due to a new feature of 1.9)
~(see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)~
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by PrettyWood** [comment]
please review

**Comment by layday** [comment]
I'm not sure if this is a good workaround. IIUC the (broader) issue with #2650 is that the transformation is no longer recursive. This will manifest in other ways (and it already has: #3544 :) It'd probably make more sense to expose `to_dict` in `json`, but adding more options always comes at a cost.

**Comment by layday** [comment]
If you don't want to delay v1.9 then reverting #2650 might be the best course of action for the time being?

**Comment by PrettyWood** [comment]
Yeah I saw new issues pop up. I will take some time tomorrow to check properly. But if no easy workaround is possible, I'll revert #2650

File: pydantic/main.py [hunk 1]
```diff
@@ -469,11 +470,12 @@ def json(
             exclude_unset = skip_defaults
         encoder = cast(Callable[[Any], Any], encoder or self.__json_encoder__)
 
-        # We don't directly call `self.dict()`, which does exactly the same thing but
-        # with `to_dict = True` because we want to keep raw `BaseModel` instances and not as `dict`.
+        # We don't directly call `self.dict()`, which is does exactly this with `to_dict=True`
```
Comment (christianbundy) [inline pydantic/main.py:473]:
```suggestion
        # We don't directly call `self.dict()`, which does exactly this with `to_dict=True`
```

**Comment by samuelcolvin** [comment]
@PrettyWood did you have time to think about this?

I've haven't dug into it in detail, but I will try to find time soon.

**Comment by christianbundy** [comment]
FWIW after the force-push this seems to solve the problem (and adds a test from #3544). At least from my perspective, this looks Fine™️.

**Comment by PrettyWood** [comment]
This should solve everything yes

**Comment by samuelcolvin** [comment]
awesome, thank you.

**Pull request merged**